app/src/main/java/com/edu/sun/oereminder/ui/timetracking/TimeTrackingFragment.kt

```diff
@@ -18,9 +28,57 @@ class TimeTrackingFragment : BaseFragment<View, TimeTrackingPresenter>(), View {
 
     override fun setupView(instanceState: Bundle?) {
 
+        toolbar.setTitle(R.string.title_item_time_tracking)
+        presenter.load()
+
+        GregorianCalendar().run {
+            btnDatePicker.text =
+                getString(
+                    R.string.date_range_format,
+                    firstDayOfMonth().toDate(DATE_FORMAT),
+                    lastDayOfMonth().toDate(DATE_FORMAT)
+                )
+            presenter.onDateSelected(firstDayOfMonth().timeInMillis, lastDayOfMonth().timeInMillis)
+        }
+
+        picker = MaterialDatePicker.Builder.dateRangePicker().apply {
+            setCalendarConstraints(
+                CalendarConstraints.Builder().setEnd(GregorianCalendar().timeInMillis).build()
+            )
+        }.build()
+
+        recyclerTimeTracking.adapter = timeTrackingAdapter
     }
 
     override fun initListener() {
 
+        btnDatePicker.setOnClickListener {
+            activity?.let { picker?.show(it.supportFragmentManager, picker.toString()) }
+        }
+
+        picker?.addOnPositiveButtonClickListener {
+            btnDatePicker.text = picker?.headerText
+            it.first?.let { from ->
+                it.second?.let { to ->
+                    presenter.onDateSelected(from, to)
+                }
+            }
```

Comment:
Nếu phải check nhiều thì không nên lồng nhau như thế 

```suggestion
            if(it.first != null && it.second != null) {
            }
```